debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java

```diff
@@ -0,0 +1,650 @@
+        final String transactionId = row.getTransactionId();
+        final Transaction transaction = getCache().get(transactionId);
+        if (transaction == null && !isRecentlyCommitted(transactionId)) {
+            // LOGGER.trace("Start: {}", row);
+            getCache().put(transactionId, new Transaction(transactionId, row.getScn(), row.getChangeTime()));
+            metrics.setActiveTransactions(getCache().size());
+        }
+    }
+
+    /**
+     * Handle processing a LogMinerEventRow for a {@code COMMIT} event.
+     *
+     * @param row the result set row
+     * @throws InterruptedException if the event dispatcher was interrupted sending events
+     */
+    protected abstract void handleCommit(LogMinerEventRow row) throws InterruptedException;
+
+    /**
+     * Handle processing a LogMinerEventRow for a {@code ROLLBACK} event.
+     * The default implementation is a no-op.
+     *
+     * @param row the result set row
+     */
+    protected void handleRollback(LogMinerEventRow row) {
+        // no-op
+    }
+
+    /**
+     * Handle processing a LogMinerEventRow for a {@code DDL} event.
+     *
+     * @param row the result set row
+     * @throws InterruptedException if the event dispatcher is interrupted sending the event
+     */
+    protected void handleSchemaChange(LogMinerEventRow row) throws InterruptedException {
+        if (hasSchemaChangeBeenSeen(row)) {
+            LOGGER.trace("DDL: Scn {}, SQL '{}' has already been processed, skipped.", row.getScn(), row.getRedoSql());
+            return;
+        }
+
+        LOGGER.trace("DDL: '{}' {}", row.getRedoSql(), row);
+        if (row.getTableName() != null) {
+            counters.ddlCount++;
+            final TableId tableId = row.getTableId();
+            dispatcher.dispatchSchemaChangeEvent(tableId,
+                    new OracleSchemaChangeEventEmitter(
+                            getConfig(),
+                            offsetContext,
+                            tableId,
+                            tableId.catalog(),
+                            tableId.schema(),
+                            row.getRedoSql(),
+                            getSchema(),
+                            row.getChangeTime(),
+                            metrics));
+        }
+    }
+
+    /**
+     * Handle processing a LogMinerEventRow for a {@code SEL_LOB_LOCATOR} event.
+     *
+     * @param row the result set row
+     */
+    protected void handleSelectLobLocator(LogMinerEventRow row) {
+        if (!getConfig().isLobEnabled()) {
+            LOGGER.trace("LOB support is disabled, SEL_LOB_LOCATOR '{}' skipped.", row.getRedoSql());
+            return;
+        }
+
+        LOGGER.trace("SEL_LOB_LOCATOR: {}", row);
+        final TableId tableId = row.getTableId();
+        final Table table = getSchema().tableFor(tableId);
+        if (table == null) {
+            LOGGER.warn("SEL_LOB_LOCATOR for table '{}' is not known, skipped.", tableId);
+            return;
```

Comment:
This should be rare assuming that correct include/exclude lists are setup.  Where this might occur would be a situation where there are no include/exclude lists defined and we're getting events from LogMiner for other PDBs or the root CDB database that has a table with LOB columns.

Reply:
ok